src/Elements/types.jl

```diff
@@ -0,0 +1,180 @@
+const HEADERS   = ["#", "##", "###", "####", "#####", "######"]
+const EMPHASIS  = ["*", "**"]
+
+const CATEGORIES = [:module,    :function, :method, :macro, :type,
+                    :typealias, :global,   :symbol, :tuple, :aside]
+
+
+## Config Nodes
+abstract ConfigNode
+
+type Config <: ConfigNode
+    objfilter               :: Any
+    style_obj               :: ASCIIString
+
+    const defaults = Dict{Symbol, Any}(
+        :objfilter                  => CATEGORIES,
+        :style_obj                  => "####",
+        )
+
+    Config(; args...) = update_config!(new(), merge(defaults, Dict(args)))
+end
+config(; cargs...) = PreConfig(; cargs...)
+
+type PreConfig <: ConfigNode
+    cargs :: Dict{Symbol, Any}
+
+    PreConfig(; cargs...) = new(Dict(cargs))
+end
```

Comment:
How necessary is the intermediate step with the ``PreConfig``? Could we go straight to a ``Config``?

Or maybe use ``PreConfig`` as the ``Config`` and just store everything in the ``Dict``...